command/agent/http_test.go

```diff
@@ -1036,18 +1049,37 @@ func TestHTTPServer_Limits_OK(t *testing.T) {
 					"error does not wrap os.ErrDeadlineExceeded: (%T) %v", err, err)
 			}
 		}
+
+		for i := 0; i < maxConns; i++ {
+			require.NoError(t, conns[i].Close())
+		}
+	}
+
+	dial := func(addr string, useTLS bool) net.Conn {
+		if useTLS {
+			cert, err := tls.LoadX509KeyPair(foocert, fookey)
+			require.NoError(t, err)
+			conn, err := tls.Dial("tcp", addr, &tls.Config{
+				Certificates:       []tls.Certificate{cert},
+				InsecureSkipVerify: true, // good enough
+			})
+			require.NoError(t, err)
```

Comment:
If we don't pass the `testing.T` from the test case into this function and it's closing over the top-level `testing.T`, I think we lose the association with the test case and it makes errors harder to track down. Might want to either pass in the test case's `testing.T` or have the function return the error and do the assertion in the caller.

Reply:
Good catch, fixed!